shared/util.js

```diff
@@ -77,6 +77,21 @@ class Util {
     return details;
   }
 
+  /**
+   * Given the entity classification dataset and a URL, identify the entity.
+   * @param {string} url
+   * @param {LH.Result.Entities=} entities
+   * @return {LH.Result.LhrEntity|string}
+   */
+  static getEntityFromUrl(url, entities) {
+    if (!entities) {
+      return Util.getLegacyRootDomain(url);
+    }
+
+    const entity = entities.find(e => e.origins.find(origin => url.startsWith(origin)));
+    return entity || Util.getLegacyRootDomain(url);
```

Comment:
how does pre-v10 LHRs play into it?   getLegacyRootDomain is used on either side of the if condition.....

is this L92 use just an extra fallback we don't expect to hit?

Reply:
We're falling back to `getLegacyRootDomain` if (1) we don't have entity classification `L87`, or (2) the URL cannot be resolved to an entity (`L92`). I think the latter shouldn't happen unless there's something wrong with the entity-classification dataset. Pre-v10 should proceed via line 88.